progs/mimc.py

```diff
@@ -0,0 +1,38 @@
+from math import log, ceil
+from honeybadgermpc.mpc import PreProcessedElements, Subgroup
+
+# ROUND: iteration time of MiMC encryption function,
+# In BLS12_381, r = ceil(log(p, 3)) = 161
+ROUND = ceil(log(Subgroup.BLS12_381))
+
+
+async def mimc_mpc(context, x, k):
```

Comment:
I think naming these methods `mimc_encrypt` and `mimc_encrypt_plain` would be better.

I think the way I would do it would be as follows--
- `mimc_encrypt` and `mimc_decrypt` functions, which are high level functions that handle both plain and shared mimc implementations.
    - These would then call on either `__mimc_encrypt_shared` or `__mimc_encrypt_plain` and `__mimc_decrypt_shared` or `__mimc_decrypt_plain` respectively based on whether or not the methods are invoked with secret-shared values or plain values.

Reply:
I think `mimc` is a function used in both encryption and decryption process. I'll have other encryption/decryption function which will call `mimc_mpc`, `mimc_plain` here.